### PR TITLE
Fixed identity generation of single node notaries as used by the driv…

### PR DESCRIPTION
**Pull request by shamsasari**
…er and MockNetwork.

The identity cert generated used to be of type SERVICE_IDENTITY when it should have been a LEGAL_IDENTITY.

### CODE REVIEW

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/IdentityGenerator.kt [hunk 1]
```diff
@@ -16,11 +19,32 @@ import java.security.cert.X509Certificate
 object IdentityGenerator {
```
Comment (kchalkias) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/IdentityGenerator.kt]: As per our discussion, methods (and class) might need the “dev” keyword in them. Some comments as well on where this is currently used and a TODO that we should agree on the actual production process.
update: Shams was faster than me in his commit!

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/IdentityGenerator.kt [hunk 2]
```diff
@@ -16,11 +19,32 @@ import java.security.cert.X509Certificate
...
+     * root cert is specified then the default one in certificates/cordadevcakeys.jks is used.
+     */
+    fun installKeyStoreWithNodeIdentity(nodeDir: Path, legalName: CordaX500Name, customRootCert: X509Certificate? = null): Party {
+        val nodeSslConfig = object : NodeSSLConfiguration {
+            override val baseDirectory = nodeDir
+            override val keyStorePassword: String = "cordacadevpass"
+            override val trustStorePassword get() = throw NotImplementedError("Not expected to be called")
+        }
+
+        val caKeyStore = loadKeyStore(javaClass.classLoader.getResourceAsStream("certificates/cordadevcakeys.jks"), "cordacadevpass")
```
Comment (kchalkias) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/IdentityGenerator.kt]: the `cordadevcakeys` (and other related Strings, like passwords) seem to exist in various places, maybe we should use a static reference if we are referring to the same object. Maybe at `SSLConfiguration` or another `DevSSLConfiguration`

**Comment by shamsasari** [comment]
As discussed, added comments on the non-prod nature of `IdentityGenerator`.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/crypto/KeyStoreWrapper.kt [hunk 3]
```diff
@@ -1,41 +1,26 @@
...
-import java.security.PublicKey
-import java.security.cert.CertPath
 import java.security.cert.Certificate
 
 class KeyStoreWrapper(private val storePath: Path, private val storePassword: String) {
     private val keyStore = storePath.read { loadKeyStore(it, storePassword) }
 
-    private fun createCertificate(serviceName: CordaX500Name, pubKey: PublicKey): CertPath {
-        val clientCertPath = keyStore.getCertificateChain(X509Utilities.CORDA_CLIENT_CA)
+    // TODO This method seems misplaced in this class
```
Comment (kchalkias) [inline node-api/src/main/kotlin/net/corda/nodeapi/internal/crypto/KeyStoreWrapper.kt]: missing comment full stop

File: node-api/src/test/kotlin/net/corda/nodeapi/internal/crypto/X509UtilitiesTest.kt [hunk 4]
```diff
@@ -266,12 +222,14 @@ class X509UtilitiesTest {
...
         val serverSocket = serverSocketFactory.createServerSocket(0) as SSLServerSocket // use 0 to get first free socket
-        val serverParams = SSLParameters(arrayOf("TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
-                "TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256",
-                "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
-                "TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA",
-                "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
-                "TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256"),
+        val serverParams = SSLParameters(
+                arrayOf(
+                        "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
```
Comment (kchalkias) [inline node-api/src/test/kotlin/net/corda/nodeapi/internal/crypto/X509UtilitiesTest.kt]: for consistency with our supported  Artemis TLS schemes (see `ArtemisTcpTransport`), let's better use only these cipher suites:
"TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
"TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
"TLS_DHE_RSA_WITH_AES_128_GCM_SHA256"

File: node-api/src/test/kotlin/net/corda/nodeapi/internal/crypto/X509UtilitiesTest.kt [hunk 5]
```diff
@@ -280,12 +238,14 @@ class X509UtilitiesTest {
...
 
         val clientSocket = clientSocketFactory.createSocket() as SSLSocket
-        val clientParams = SSLParameters(arrayOf("TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
-                "TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA256",
-                "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
-                "TLS_ECDH_ECDSA_WITH_AES_128_CBC_SHA",
-                "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
-                "TLS_ECDH_ECDSA_WITH_AES_128_GCM_SHA256"),
+        val clientParams = SSLParameters(
+                arrayOf(
```
Comment (kchalkias) [inline node-api/src/test/kotlin/net/corda/nodeapi/internal/crypto/X509UtilitiesTest.kt]: ditto

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/IdentityGenerator.kt (hunk 2 above)
Comment (shamsasari) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/IdentityGenerator.kt]: Yes, I have another branch where I'm cleaning up our usage of key stores. I'll add constants for these dev passwords in there.

File: node-api/src/main/kotlin/net/corda/nodeapi/internal/IdentityGenerator.kt [hunk 6]
```diff
@@ -32,35 +63,36 @@ object IdentityGenerator {
...
      *
-     * @param dirs List of node directories to place the generated identity and key pairs in.
+     * @param nodeDirs List of node directories to place the generated identity and key pairs in.
      * @param name The name of the identity.
      * @param threshold The threshold for the generated group [CompositeKey].
      * @param customRootCert the certificate to use as the Corda root CA. If not specified the one in
      *      internal/certificates/cordadevcakeys.jks is used.
      */
-    private fun generateToDisk(dirs: List<Path>,
+    private fun generateToDisk(nodeDirs: List<Path>,
```
[the remark this answers is not in the document]
Comment (shamsasari) [reply node-api/src/main/kotlin/net/corda/nodeapi/internal/IdentityGenerator.kt]: Yep, forgot to do that.

**Comment by shamsasari** [comment]
Comments addressed.

**Pull request merged**